FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/HandlePasswordProtectedMsgWorker.kt

```diff
@@ -0,0 +1,425 @@
+              msgEntity = msgEntity.copy(isEncrypted = false),
+              atts = attachments
+            )
+
+            //get recipients that will be used to create password-encrypted msg
+            val toCandidates = mimeMsgWithAttachments.getMatchingRecipients(
+              type = Message.RecipientType.TO,
+              list = GeneralUtil.getRecipientsWithoutUsablePubKeys(
+                context = applicationContext,
+                emails = mimeMsgWithAttachments.getAddresses(Message.RecipientType.TO)
+              )
+            )
+
+            val ccCandidates = mimeMsgWithAttachments.getMatchingRecipients(
+              type = Message.RecipientType.CC,
+              list = GeneralUtil.getRecipientsWithoutUsablePubKeys(
+                context = applicationContext,
+                emails = mimeMsgWithAttachments.getAddresses(Message.RecipientType.CC)
+              )
+            )
+
+            val bccCandidates = mimeMsgWithAttachments.getMatchingRecipients(
+              type = Message.RecipientType.BCC,
+              list = GeneralUtil.getRecipientsWithoutUsablePubKeys(
+                context = applicationContext,
+                emails = mimeMsgWithAttachments.getAddresses(Message.RecipientType.BCC)
+              )
+            )
+
+            if (toCandidates.isNotEmpty()) {
+              //start of creating and uploading a password-protected msg to FES
+              val fromAddress = mimeMsgWithAttachments.getFromAddress()
+              val domain = EmailUtil.getDomain(fromAddress)
+              val idToken = getGoogleIdToken()
+              val replyToken = fetchReplyToken(apiRepository, domain, idToken)
+              val messageUploadRequest = MessageUploadRequest(
+                associateReplyToken = replyToken,
+                from = fromAddress,
+                to = toCandidates.map { (it as InternetAddress).address },
+                cc = ccCandidates.map { (it as InternetAddress).address },
+                bcc = bccCandidates.map { (it as InternetAddress).address }
+              )
+
+              //prepare bodyWithReplyToken
+              val replyInfo = Base64.getEncoder().encodeToString(
+                GsonBuilder().create().toJson(messageUploadRequest).toByteArray()
+              )
+
+              val infoDiv = genInfoDiv(replyInfo)
+              val originalText = getDecryptedContentFromMessage(
+                mimeMsgWithAttachments,
+                accountSecretKeys,
+                keysStorage
+              )
+              val bodyWithReplyToken = originalText + "\n\n" + infoDiv
+
+              // construct a regular plain mime message using bodyWithReplyToken + attachments
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.TO, toCandidates)
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.CC, ccCandidates)
+              mimeMsgWithAttachments.setRecipients(Message.RecipientType.BCC, bccCandidates)
```

Comment:
This should be done right where `mimeMsgWithAttachments` (`plainMimeMsgWithAttachments`) is created, it belongs together. It's not good to do a little bit of work above and a bit more work down here if it's related. 

Actually, together with the `createMimeMsg`, and also adding body part + writing to output stream - that should all be on a separate method together, so that it can just return `rawMimeMsg` but name the result `rawPlainMimeMsgWithAttachments`

The method would be called `createRawPlainMimeMsgWithAttachments()`.